go-controller/pkg/node/gateway_localnet.go

```diff
@@ -239,89 +246,374 @@ func localnetIPTablesHelper(subnet *net.IPNet) (util.IPTablesHelper, error) {
 	return ipt, nil
 }
 
-func localnetIptRules(svc *kapi.Service, gatewayIP string) []iptRule {
-	rules := make([]iptRule, 0)
-	for _, svcPort := range svc.Spec.Ports {
-		protocol, err := util.ValidateProtocol(svcPort.Protocol)
+type activeSocket interface {
+	Close() error
+}
+
+type localNodePortWatcher interface {
```

Comment:
Does this need to be an interface? Can't this just be a single struct?

Reply:
It's been removed, as per my previous comment for the refactor.